src/structures/member.ts

```diff
@@ -153,6 +153,26 @@ export class Member extends SnowflakeBase {
     )
   }
 
+  /**
+   * Sets Timeout for the Member
+   * @param expiration Value to set
+   */
+  async setTimeout(expiration?: Date, reason?: string): Promise<Member> {
+    return await this.edit(
+      {
+        communicationDisabledUntil: expiration
+      },
+      reason
+    )
+  }
+
+  /**
+   * Resets Timeout for the Member
+   */
+  async resetTimeout(reason?: string): Promise<Member> {
+    return await this.setTimeout(undefined, reason)
```

Comment:
use `null` to remove timeout

Reply:
member.communicationDisabledUntil is Date | undefined, it's not nullable

Reply:
huh, then it must be implemented wrong
from [this](https://discord.com/developers/docs/resources/guild#modify-guild-member-json-params) it is nullable

Reply:
huh, that's pretty strange, does it function right now?

Reply:
yes, even the official discord client uses `null` to remove timeout

Reply:
i mean does harmony's impl work, since its wrong

Reply:
oh, probably not because `undefined` means that we don't want to change the value, not removing it

but again it's just `MemberData` type that is wrong, so changing the type ~~or do `@ts-ignore`~~ will work

Reply:
Right, we forgot to add `| null` to `MemberData`'s `communication_disabled_until`. Can you do that, and change this to null?